src/unix_permissions.rs

```diff
@@ -1,22 +1,43 @@
 use std::io;
+use libc::chmod;
+use std::ffi::CString;
+use std::io::Error;
 
 /// A NOOP struct for bringing the API between Windows and Unix up to parity. To set permissions
 /// properly on Unix, you can just use `std::os::unix::fs::PermissionsExt`.
-pub struct SecurityAttributes;
+pub struct SecurityAttributes {
+    mode: Option<u32>
+}
 
 impl SecurityAttributes {
     /// New default security attributes.
     pub fn empty() -> Self {
-        SecurityAttributes
+        SecurityAttributes {
+            mode: None
+        }
     }
 
     /// New security attributes that allow everyone to connect.
-    pub fn allow_everyone_connect() -> io::Result<Self> {
-        Ok(SecurityAttributes)
+    pub fn allow_everyone_connect(mut self, mode: Option<u32>) -> io::Result<Self> {
```

Comment:
why do you need `mode` argument for `allow_everyone_connect` - why not  just 777 if it is for everyone?

to set specific permissions, probably implement another method, which have another name

Reply:
hmm, yeah you're right will add another method.